### ISSUE TITLE
Network instability leads to program failing

### ISSUE THREAD

**Issue by ghost** [agytw/stock-crawler#5]
Ocassionally happens, the error logs are pasted below

```
Traceback (most recent call last):
  File "main.py", line 84, in <module>
    one_day-date, three_Day_date = main_action(one_Day_date, three_Day-date)
  File "main.py", line 32, in main_action
    decoded_item = basics.decode(ticker)
  File "[masked]/crawler/basics.py", line 18, in https_get
    item = https_get(url)
  File "[masked]/crawler/basics.py", line 12, in https_get
    response = response.read().decode(encoding='UTF-8').strip()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

**Comment by ghost** [comment]
Might be a network instability

### Solution:
re-send the request

```python
for _ in range(max_retries):
    try:
        # send the request
        break
    except UnicodeDecodeError:
        pass
```

**Comment by ghost** [comment]
Fixed in ef15a81eb93c8c77490a1accc5941c7fd1b78c0b and b7e23522cac461869746bf2f23bcbc5ab61e1442
